test/core/test/expect.test.ts

```diff
@@ -346,3 +346,42 @@ describe('iterator', () => {
     expect(a).not.toStrictEqual(b)
   })
 })
+
+describe('Temporal equality', () => {
```

Comment:
Thanks for PR! I think we should test against the polyfill which is supposed to be spec compliant.